### DISCUSSION TITLE
Securing ZOAU - Best Practices

### DISCUSSION THREAD

**Discussion by EnsonoStevenPerva** [IBM/zoau-samples#30] [Ideas] [answered]
I was just reviewing the installation information for ZOAU and wanted to see if I could start a discussion around the section of the documentation found [here](https://www.ibm.com/docs/en/zoau/1.2.0?topic=installing-configuring-zoa-utilities) titled "_Protecting ZOA Utilities for its APF authorized components_". Could we generate (or at least start a dialogue on) a best-practices recommendation on what would be the most appropriate way to control these components? This could maybe evolve into a security hardening guide for ZOAU.

**Comment by MikeFultonDev** [comment]
That's a great idea. It would be good to discuss the best way to cover this.

Regarding the APF authorized code - please note that the code does not 'do' anything itself that requires APF authorization - it is required because it loads modules that are APF authorized in the same LE enclave. 
In particular, the code for mvscmd and mvscmdauth is the same binary - the only difference is that mvscmdauth is APF authorized so that it can load APF authorized code (like IKJEFT01) to run it.

**Comment by EnsonoStevenPerva** [comment]
There are several helpers that are APF authorized, would restricting their use to a privileged group of users prevent their counterparts from being functional for a general user? I'm specifically a fan of using `jls` and `ddls`. I think the perfect case scenario would be to simply remove the ability for anyone to execute `mvscmdauth` in favor of the user and group permission classes.

I think understanding the risk of allowing any-old-user to execute `mvscmdauth` would be a strong first step. I personally don't understand the risk entirely beyond potentially executing a program that escalates privileges...

**Comment by MikeFultonDev** [comment, marked as answer]
Let me add some background on mvscmd and mvscmdauth.
mvscmd and mvscmdauth are the same code. One is a link of the other, and mvscmdauth has the 'APF authorized' bit set on so it will be loaded as authorized. 
mvscmd can ONLY run unauthorized code. If you try to run an authorized program (e.g. IKJEFT01) you will get an error.
mvscmdauth can ONLY run authorized code. If you try to run an unauthorized program (e.g. IEBCOPY) you will get an error.
Other than the code to correctly detect if mvscmdauth is itself running authorized, and the ability to load an authorized program and call it, mvscmdauth is not doing anything that requires authorization. I view mvscmdauth as being analogous to a JES program, which also would need to be authorized for it to be able to run JCL that then loads particular authorized or unauthorized programs and then runs them. 

Similarly, the code to run JES services (jls, ddls, pjdd, jcan, jsub) needs to be authorized for the same reason - they need to run authorized code, but again, they do not do anything requiring authorization other than calling these services. 

Hope this answer helps clear things up a bit.

**Reply by EnsonoStevenPerva** [reply]
This is tremendously helpful! Thanks for making that clear, @MikeFultonDev!
